src/xtgeo/cube/_cube_import.py

```diff
@@ -20,103 +48,105 @@ def import_segy(sfile: xtgeo._XTGeoFile) -> Dict:
     """Import SEGY via the SegyIO library.
 
     Args:
-        sfile (str): File name of SEGY file
+        sfile: File object for SEGY file
     """
-    # pylint: disable=too-many-statements
-    # pylint: disable=too-many-locals
     sfile = sfile.file
 
-    logger.debug("Inline sorting %s", segyio.TraceSortingFormat.INLINE_SORTING)
+    attributes = dict()
+
+    try:
+        # cube with all traces present
+        with segyio.open(sfile, "r") as segyfile:
+            attributes = _import_segy_all_traces(segyfile)
+    except ValueError as verr:
+        if any([word in str(verr) for word in ["Invalid dimensions", "inconsistent"]]):
+            # cube with missing traces
+            print(
+                "XTGeo notice: Missing or inconsistent traces in SEGY, will try "
+                "but check result carefully!"
+            )
 
-    with segyio.open(sfile, "r") as segyfile:
-        segyfile.mmap()
+            with segyio.open(sfile, "r", ignore_geometry=True) as segyfile:
+                attributes = _import_segy_incomplete_traces(segyfile)
+        else:
+            raise
+    except BaseException as berr:
+        print(f"Cannot parse SEGY file: {str(berr)}")
+        raise
 
-        values = segyio.tools.cube(segyfile)
+    if not attributes:
+        raise RuntimeError("Could not get attributes for segy file")
 
-        logger.info(values.dtype)
-        if values.dtype != np.float32:
-            xtg.warnuser(
-                "Values are converted from {} to {}".format(values.dtype, "float32")
-            )
+    attributes["segyfile"] = sfile
+    return attributes
+
+
+def _import_segy_all_traces(segyfile: segyio.segy.SegyFile) -> Dict:
+    """Import a a full cube SEGY via the SegyIO library to xtgeo format spec.
+
+    Here, the segyio.tools.cube function can be applied
+
+    Args:
+        segyfile: Filehandle from segyio
+    """
+    segyfile.mmap()
+
+    values = _validate_cube_values(segyio.tools.cube(segyfile))
+
+    ilines = segyfile.ilines
+    xlines = segyfile.xlines
+
+    ncol, nrow, nlay = values.shape
+
+    # get additional but required geometries for xtgeo; xori, yori, xinc, ..., rotation
+    attrs = _segy_all_traces_attributes(segyfile, ncol, nrow, nlay)
+
+    attrs["ilines"] = ilines
+    attrs["xlines"] = xlines
+    attrs["values"] = values
+
+    return attrs
+
+
+def _validate_cube_values(values: np.ndarray) -> np.ndarray:
```

Comment:
Consider renaming this? I was a bit thrown off by a validate function returning `values`, but not very important, the function is easy to read.

Reply:
Changed to `_process_cube_values()`